src/test/ui/mir/issue-89485.rs

```diff
@@ -0,0 +1,18 @@
+// Regression test for issue #89485.
```

Comment:
I think it would be a good idea to add a copy of this test with `-Zunsound-mir-opts` enabled